src/tx/builder/field-types.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  writeId, readId, isNameValid, produceNameId, ensureNameValid, getMinimumNameFee, readInt, writeInt
+} from './helpers'
+import { InsufficientNameFeeError, IllegalArgumentError } from '../../utils/errors'
+import BigNumber from 'bignumber.js'
+
+/**
+ * Class implementing a common interface for all TX field types
+ * @class
+ */
+export class Field {
+  public prototype: any
+
+  /**
+   * Serialize value to Buffer
+   * @function
+   * @param {String | BigNumber} value
+   * @param {Object} _options
+   * @return {Buffer}
+   */
+  static serialize (value: string | BigNumber, _options: object = {}): Buffer {
+    return Buffer.from(value)
+  }
+
+  /**
+   * Deserialize value from Buffer
+   * @function
+   * @param {Buffer} value
+   * @return {String}
+   */
+  static deserialize (value: Buffer): string {
+    return value.toString()
+  }
+}
+
+/**
+ * Name Field Type Class
+ * @class
+ * @extends Field
+ */
+export class Name extends Field {
+  /**
+   * Serialize AENS name to Buffer
+   * @param value - AENS name
+   */
+  static serialize (value: string): Buffer {
+    ensureNameValid(value)
```

Comment:
```suggestion
  static serialize (value: `${string}.chain`): Buffer {
```
but `${string}.chain` should be extracted somewhere